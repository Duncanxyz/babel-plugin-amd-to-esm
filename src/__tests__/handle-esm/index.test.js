const babel = require('@babel/core');
const plugin = require('../../index.js');

const fs = require('fs');
const path = require('path');
const mock = require('mock-fs');

it('handle esm', () => {
  mock({
    code: mock.load(path.resolve(__dirname, './code')),
    node_modules: mock.load(path.resolve(__dirname, '../../../node_modules')),
  });

  const rootDir = path.resolve('./code');

  const originCode = fs.readFileSync(
    path.resolve(rootDir, './wrapper/amd-a.js'),
    'utf8'
  );
  const { code } = babel.transform(originCode, {
    filename: path.resolve(rootDir, './wrapper/amd-a.js'),
    plugins: [
      [
        plugin,
        {
          handleESMDependencies: true,
          rootDir: rootDir,
          alias: {
            '@': rootDir,
          },
        },
      ],
    ],
  });
  expect(code).toMatchSnapshot();

  mock.restore();
});
