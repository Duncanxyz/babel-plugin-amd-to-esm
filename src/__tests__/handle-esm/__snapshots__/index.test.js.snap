// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle esm 1`] = `
"import _amdDep from "../amd-b.js";
import { esmA1, default as esmDefault, esmA2 as esmAA2 } from "../esm-a.js";
import * as esmB from "@/esm-b";
import VueComponent from "./VueComponent";
import ReactComponent from "../ReactComponent";
const {
  amdB1,
  default: amdB2
} = _amdDep;
export default {
  name: 'amd-a'
};"
`;
